components/PsychrometricCharts.py

```diff
@@ -0,0 +1,177 @@
+    return 610.94 * np.exp(17.625 * temp_c / (temp_c + 243.04))
+
+def calculate_dew_point(temp_c, relative_humidity):
+    saturation_vapor_pressure = calculate_saturation_vapor_pressure(temp_c)
+    actual_vapor_pressure = relative_humidity / 100.0 * saturation_vapor_pressure
+    dew_point = 243.04 * np.log(actual_vapor_pressure / 610.94) / (17.625 - np.log(actual_vapor_pressure / 610.94))
+    return dew_point
+
+def calculate_wet_bulb_temp(dry_bulb_temp, relative_humidity):
+    epsilon = 0.0001
+    wet_bulb_temp = dry_bulb_temp
+    while True:
+        prev_wet_bulb_temp = wet_bulb_temp
+        wet_bulb_temp = dry_bulb_temp * math.atan(0.151977 * math.sqrt(relative_humidity + 8.313659)) \
+                        + math.atan(dry_bulb_temp + relative_humidity) \
+                        - math.atan(relative_humidity - 1.676331) \
+                        + 0.00391838 * (relative_humidity ** 1.5) * math.atan(0.023101 * relative_humidity) \
+                        - 4.686035
+        if abs(prev_wet_bulb_temp - wet_bulb_temp) < epsilon:
+            break
+    return wet_bulb_temp
+
+def calculate_absolute_humidity(dry_bulb_temp, relative_humidity):
+    saturation_vapor_pressure = calculate_saturation_vapor_pressure(dry_bulb_temp)
+    actual_vapor_pressure = relative_humidity / 100.0 * saturation_vapor_pressure
+    absolute_humidity = 216.7 * (actual_vapor_pressure / (dry_bulb_temp + 273.15))
+    return absolute_humidity
+
+def calculate_humidity_parameters(dry_bulb_temp, relative_humidity):
+    dew_point_temp = calculate_dew_point(dry_bulb_temp, relative_humidity)
+    wet_bulb_temp = calculate_wet_bulb_temp(dry_bulb_temp, relative_humidity)
+    absolute_humidity = calculate_absolute_humidity(dry_bulb_temp, relative_humidity)
+    enthalpy = (dry_bulb_temp + relative_humidity) * 0.24  
+    return {
+        "Dry Bulb Temperature (°C)": dry_bulb_temp,
+        "Relative Humidity (%)": relative_humidity,
+        "Absolute Humidity (g/m^3)": absolute_humidity,
+        "Wet Bulb Temperature (°C)": wet_bulb_temp,
+        "Dew Point Temperature (°C)": dew_point_temp,
+        "Enthalpy (kJ/kg)": enthalpy  
+    }
+
+def generate_vapor_pressure_chart(fixed_temperature, fixed_humidity):
+    temperatures = np.linspace(10, 36, 500)
+    relative_humidities = np.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 100])
+    
+    traces = []
+
+
+    green_zone_1_left = 22
+    green_zone_1_right = 24
+    green_zone_2_left = 24
+    green_zone_2_right = 26
+    green_zone_3_left = 26
+    green_zone_3_right = 28
+
+    vapor_pressure_min = 0  
+    vapor_pressure_max = 3000
+
+    traces.append(go.Scatter(
+        x=np.concatenate([np.array([green_zone_1_left, green_zone_1_right]), 
+                          np.array([green_zone_1_right, green_zone_1_left])]),
+        y=np.concatenate([np.array([vapor_pressure_max, vapor_pressure_max]), 
+                          np.array([vapor_pressure_min, vapor_pressure_min])]),
+        fill='toself',
```

Comment:
reformat the code with Black

Reply:
got it ,sir

Reply:
![image](https://github.com/user-attachments/assets/e03eadcf-70ad-4e79-8ff4-5ec01b973c02)
i already black the code,sir